pkg/client/services.go

```diff
@@ -77,7 +77,13 @@ func (c *services) Get(name string) (result *api.Service, err error) {
 // Create creates a new service.
 func (c *services) Create(svc *api.Service) (result *api.Service, err error) {
 	result = &api.Service{}
-	err = c.r.Post().Namespace(c.ns).Resource("services").Body(svc).Do().Into(result)
+	// v1beta3 does not allow POST without a namespace.
+	needNamespace := !api.PreV1Beta3(c.r.APIVersion())
```

Comment:
The RESTStorage objects and the api_installer now properly check a lot of the conditions. If there is client code that is allowing Namespace to be wrong when it passes to the Services code, we should either fix the client code or reject it on the server (the standard api/rest.go code should do that now).

Opened https://github.com/GoogleCloudPlatform/kubernetes/pull/5738 to clean some of this up.

----- Original Message -----
> > @@ -77,7 +77,13 @@ func (c *services) Get(name string) (result
> > *api.Service, err error) {
> >  // Create creates a new service.
> >  func (c *services) Create(svc *api.Service) (result *api.Service, err
> >  error) {
> >  	result = &api.Service{}
> > -	err =
> > c.r.Post().Namespace(c.ns).Resource("services").Body(svc).Do().Into(result)
> > +	// v1beta3 does not allow POST without a namespace.
> > +	needNamespace := !api.PreV1Beta3(c.r.APIVersion())
> 
> ok. What do you mean by "normalizing the server and the client"?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/5587/files#r26881133